### PR TITLE
Hw hb 06

### PR DESCRIPTION
**Pull request by YuraMitriakhin**

### CODE REVIEW

**Review by IwalkInSilence — APPROVED** [review]
Good job!

File: src/main/java/com/gmail/yuramitryahin/Main.java [hunk 1]
```diff
@@ -2,19 +2,36 @@
...
+    private static final AuthenticationService authenticationService
+            = (AuthenticationService) injector.getInstance(AuthenticationService.class);
+    private static final ShoppingCartService shoppingCartService = (ShoppingCartService) injector
+            .getInstance(ShoppingCartService.class);
+    private static final MovieSessionService movieSessionService = (MovieSessionService) injector
+            .getInstance(MovieSessionService.class);
+    private static final MovieService movieService = (MovieService) injector
+            .getInstance(MovieService.class);
+    private static final User user = authenticationService
+            .register("user@gmail.com", "1234");
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/Main.java]: let's make this a local variable

File: src/main/java/com/gmail/yuramitryahin/model/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import javax.persistence.MapsId;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "shoping_cart")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/model/ShoppingCart.java]: do we need this?

File: src/main/java/com/gmail/yuramitryahin/model/ShoppingCart.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    private long id;
+    @OneToMany
+    private List<Ticket> tickets;
+
+    @OneToOne(cascade = CascadeType.MERGE)
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/model/ShoppingCart.java]: let's remove this cascade?

File: src/main/java/com/gmail/yuramitryahin/model/Ticket.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "ticket")
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @OneToOne
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/model/Ticket.java]: are you sure that here should be 1-1 relationship?

File: src/main/java/com/gmail/yuramitryahin/model/Ticket.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    @JoinColumn(name = "movie_session_id")
+    private MovieSession movieSession;
+
+    @OneToOne
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/model/Ticket.java]: same here

File: src/main/java/com/gmail/yuramitryahin/service/AuthenticationServiceImpl.java [hunk 6]
```diff
@@ -25,6 +27,10 @@ public User login(String email, String password) throws AuthenticationException
 
     @Override
     public User register(String email, String password) {
-        return userService.add(new User(email, password));
+        User user = new User(email, password);
+        userService.add(user);
+        System.out.println(user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/service/AuthenticationServiceImpl.java]: let's remove this line

File: src/main/java/com/gmail/yuramitryahin/service/ShoppingCartServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/service/ShoppingCartServiceImpl.java]: if there is no shopping cart for the user - we still save ticket to the db, is it ok?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/gmail/yuramitryahin/Main.java (hunk 1 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/Main.java]: but if I make local, check style has a problem and write that user must be final

File: src/main/java/com/gmail/yuramitryahin/model/ShoppingCart.java (hunk 2 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/model/ShoppingCart.java]: fixed

File: src/main/java/com/gmail/yuramitryahin/model/ShoppingCart.java (hunk 3 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/model/ShoppingCart.java]: fixed

File: src/main/java/com/gmail/yuramitryahin/model/Ticket.java (hunk 4 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/model/Ticket.java]: oops

File: src/main/java/com/gmail/yuramitryahin/model/Ticket.java (hunk 5 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/model/Ticket.java]: fixed

File: src/main/java/com/gmail/yuramitryahin/service/AuthenticationServiceImpl.java (hunk 6 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/service/AuthenticationServiceImpl.java]: fixed

File: src/main/java/com/gmail/yuramitryahin/service/ShoppingCartServiceImpl.java (hunk 7 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/service/ShoppingCartServiceImpl.java]: not ok, fixed

File: src/main/java/com/gmail/yuramitryahin/Main.java (hunk 1 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/Main.java]: I made local final variable

File: src/main/java/com/gmail/yuramitryahin/Main.java [hunk 8]
```diff
@@ -2,19 +2,36 @@
...
-        System.out.println(authenticationService.register(user.getEmail(), user.getPassword()));
-        System.out.println(authenticationService.login(user.getEmail(), user.getPassword()));
+        final User user = authenticationService
+                .register("user@gmail.com", "1234");
+        MovieSession movieSession = new MovieSession();
+        Movie movie = new Movie();
+        movie.setTitle("Moana");
+        movieSession.setMovie(movie);
+        movieService.add(movie);
+        movieSessionService.add(movieSession);
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/Main.java]:
```suggestion
        MovieSession movieSession = new MovieSession();
        Movie movie = new Movie();
        movie.setTitle("Moana");
        movieSession.setMovie(movie);
        movieService.add(movie);
        movieSessionService.add(movieSession);
        User user = authenticationService
                .register("user@gmail.com", "1234");
```
can we do like this?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/gmail/yuramitryahin/Main.java (hunk 8 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/Main.java]: yep, fixed, but can you explain the difference?

Comment (KseniiaMakarova) [reply src/main/java/com/gmail/yuramitryahin/Main.java]: there is no difference, except that now the checkstyle passes without making the variable final

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/gmail/yuramitryahin/Main.java (hunk 8 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/Main.java]: ok)

**Pull request merged**